PebbleSharp.Core/AppMessage/AppMessagePacket.cs

```diff
@@ -0,0 +1,524 @@
+            set
+            {
+                if (value.Length == Length)
+                {
+                    Value = BitConverter.ToInt32(value, 0);
+                }
+                else
+                {
+                    throw new InvalidOperationException("Incorrect # of bytes");
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            return Value.ToString();
+        }
+    }
+
+    public class AppMessageString : AppMessageDictionaryEntry<string>
+    {
+        public override PackedType PackedType
+        {
+            get { return PackedType.String; }
+        }
+
+        public override ushort Length
+        {
+            get { return (ushort)ValueBytes.Length; }
+        }
+
+        public override string Value
+        {
+            get
+            {
+                return base.Value;
+            }
+            set
+            {
+                if (value != null && value.EndsWith("\0"))
+                {
+                    base.Value = value.Substring(0, value.Length - 1);
+                }
+                else
+                {
+                    base.Value = value;
+                }
+            }
+        }
+
+        public override byte[] ValueBytes
+        {
+            get
+            {
+                return System.Text.UTF8Encoding.UTF8.GetBytes(Value+"\0");
+            }
+            set
+            {
+                if (value.Length <= ushort.MaxValue)
+                {
+                    Value = System.Text.UTF8Encoding.UTF8.GetString(value,0,value.Length);
+                }
+                else
+                {
+                    throw new OverflowException("Specified string is too large for length to fit in a ushort");
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            return Value.ToString();
+        }
+    }
+
+    public class AppMessageBytes : AppMessageDictionaryEntry<byte[]>
+    {
+        public override PackedType PackedType
+        {
+			get { return PackedType.Bytes; }
+        }
+
+        public override ushort Length
+        {
+            get { return (ushort)Value.Length; }
+        }
+
+        public override byte[] ValueBytes
+        {
+            get { return Value; }
+            set
+            {
+                if (value.Length <= ushort.MaxValue)
+                {
+                    Value = value;
+                }
+                else
+                {
+                    throw new OverflowException("Specified array is too large for length to fit in a ushort");
+                }
+            }
+        }
+
+        public override string ToString()
+        {
+            var s = new StringBuilder();
+            foreach(var b in Value)
+            {
+                s.Append(b.ToString());
+                s.Append(",");
+            }
+            return s.ToString();
+        }
```

Comment:
I had a .ToHexString() method in my other fork that I used for debugging, I think I'd rather use that here since hex is going to be easier to debug and that's really the main use of ToString in this case, thoughts?